blockchain/processing.py

```diff
@@ -255,6 +299,8 @@ def _execute_phase_1(self, config, current_block_id):
             # update status of approved transactions
             for tx in approved_transactions:
                 tx["header"]["status"] = "approved"
+                if tx["header"]["transaction_type"] == "TT_SUB_REQ":
+                    self.smart_contracts["TT_SUB_REQ"](tx)
```

Comment:
I think that lines 302 & 303 will not be needed as the `TT_SUB_REQ` smart contract will have been called due to the above requested refactoring.